File: python/easy/implement_strstr.py
def strtr(haystack, needle):
  if len(needle) == 0:
    print(0)
  elif needle in haystack:
    for i in range(0, len(haystack)):
      if haystack[i:i+len(needle)] == needle:
         print(i)
  else:
    print(-1)

h = "hello"
n = "o"
strtr(h, n)

File: javascript/easy/implement_strstr.js
// 28. Implement strStr()
// https://leetcode.com/problems/implement-strstr/

var strStr = function(haystack, needle) {
    if (needle.length == 0) {
        return 0;
    } else if (haystack.includes(needle)) {
        for (var i = 0; i <= haystack.length; i++) {
            if (haystack.substring(i, i + needle.length) == needle) {
                return i;
            }
        }
    } else {
        return -1;
    }
};

var h = "hello";
var n = "ll";
console.log(strStr(h, n));